### ISSUE TITLE
[Bug]: AutoImport mode generates conflicting imports

### ISSUE THREAD

**Issue by Zuplyx** [INRIA/spoon#5121]
### Describe the bug

In AutoImport mode Spoon generates imports for every fully qualified reference, even if the resulting import statements conflict.

### Source code you are trying to analyze/transform

```Java
public class ImportConflicts
{

  public java.sql.Date convert(java.util.Date date)
  {
    // perform some conversion from java.util.Date to java.sql.Date
    return new java.sql.Date(date.getTime());
  }

}
```

### Source code for your Spoon processing

```Java
import spoon.Launcher;
import spoon.SpoonAPI;

public class Refactoring
{

  /**
   * Starts spoon.
   *
   * @param args First arg is source input directory.
   */
  public static void main(String[] args)
  {
    SpoonAPI spoon = new Launcher();
    spoon.getEnvironment().setAutoImports(true);
    spoon.addInputResource(args[0]);
    spoon.run();
  }
}
```

### Actual output

```Java
import java.sql.Date;
import java.util.Date;
public class ImportConflicts {
    public Date convert(Date date) {
        // perform some conversion from java.util.Date to java.sql.Date
        return new Date(date.getTime());
    }
}
```

### Expected output

```Java
Spoon detects the conflict and leaves at least one of the references in the fully qualified form.
```

### Spoon Version

Tested with 10.2.0 and 10.3.0-beta-16

### JVM Version

11

### What operating system are you using?

Tested on Windows 10 and Linux

**Comment by I-Al-Istannen** [comment]
I stumbled upon this as well a few days ago and have an open PR for this: https://github.com/INRIA/spoon/pull/5119

**Comment by Zuplyx** [comment]
Fixed with version 10.3.0
